data/store.go

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (C) 2023 Nuts community
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+package data
+
+import (
+	"context"
+	"time"
+)
+
+// Store is an in-memory store that contains a mapping from transaction signer to its controller.
+// It also contains three sliding windows with length and resolution of: (1 hour, 1 minute), (1 day, 1 hour), (30 days, 1 day).
+// A transaction can be added, the store will resolve the signer and the controller of the signer.
+type Store struct {
+	mapping        map[string]string
+	slidingWindows []slidingWindow
+}
+
+func NewStore() *Store {
+	s := &Store{
+		mapping: make(map[string]string),
+		slidingWindows: []slidingWindow{
+			{
+				resolution:       time.Minute,
+				length:           time.Hour,
+				evictionInterval: time.Second,
+			},
+			{
+				resolution:       time.Hour,
+				length:           24 * time.Hour,
+				evictionInterval: time.Minute,
+			},
+			{
+				resolution:       24 * time.Hour,
+				length:           30 * 24 * time.Hour,
+				evictionInterval: time.Minute,
+			},
+		},
+	}
+
+	// initialize all windows with empty dataPoints using the init function
+	for i := range s.slidingWindows {
+		s.slidingWindows[i].init()
```

Comment:
`init` is ugly and easy to forget. a `newSlidingWindow(resolution, length, interval time.Duration)` should include the init logic